bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,314 @@
+
+        boolean needsUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, ZoneId.of("UTC")))
+                || responseMap == null || needToFetchHistoricDays(true);
+
+        boolean hasNextDayValue = needsUpdate ? false
+                : responseMap.entrySet().stream()
+                        .anyMatch(x -> x.getKey().getDayOfYear() == currentUtcTime().plusDays(1).getDayOfYear());
+        boolean readyForNextDayValue = needsUpdate ? true
+                : currentUtcTime().isAfter(currentUtcTime().withHour(config.spotPricesAvailableUtcHour));
+
+        // Update whole time series
+        if (needsUpdate || (!hasNextDayValue && readyForNextDayValue)) {
+            logger.debug("Updating timeseries");
+            Request request = new Request(config.securityToken, config.area, startUtc, endUtc);
+            Client client = new Client();
+            boolean success = false;
+
+            try {
+                responseMap = client.doGetRequest(request, 30000);
+                TimeSeries baseTimeSeries = new TimeSeries(EntsoeBindingConstants.TIMESERIES_POLICY);
+                for (Map.Entry<ZonedDateTime, Double> entry : responseMap.entrySet()) {
+                    BigDecimal exchangedKwhPrice = getExchangedKwhPrice(entry.getValue());
+                    State baseState = new QuantityType<>(exchangedKwhPrice + " " + baseUnit.getName() + "/kWh");
+                    baseTimeSeries.add(entry.getKey().toInstant(), baseState);
+                }
+                lastDayAheadReceived = currentUtcTime();
+                sendTimeSeries(EntsoeBindingConstants.CHANNEL_SPOT_PRICES, baseTimeSeries);
+                updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+                scheduler.schedule(this::triggerChannelSpotPricesReceived, 30, TimeUnit.SECONDS);
+                success = true;
+            } catch (EntsoeResponseException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                logger.error("{}", e.getMessage());
+            } catch (EntsoeConfigurationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                logger.error("{}", e.getMessage());
+            } finally {
+                schedule(success);
+            }
+        }
+        // Update current hour with refreshed exchange rate
+        else {
+            logger.debug("Updating current hour");
+            updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+            schedule(true);
+        }
+    }
+
+    private void schedule(boolean success) {
+        logger.trace("schedule(success:{})", success);
+        if (!success) {
+            refreshJob = scheduler.schedule(this::refreshPrices, 5, TimeUnit.MINUTES);
+        } else {
+            refreshJob = scheduler.schedule(this::refreshPrices, getSecondsToNextHour(), TimeUnit.SECONDS);
```

Comment:
If you want a slightly faster/more accurate update, you could consider calculating in milliseconds instead. See for example:
https://github.com/openhab/openhab-addons/blob/104f32892e87d3c5236dad7326d10cad4b15f6df/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L691-L696